src/components/calendar/calendar.jsx

```diff
@@ -0,0 +1,199 @@
+  ) return styles['highlighted-date'];
+
+  if (iterator.getMonth() === month) return styles.date;
+
+  return styles['not-current-date'];
+}
+
+function getCell(iterator, month, highlightedDate) {
+  const date = iterator.getDate();
+  const className = getCellClassName(iterator, month, highlightedDate);
+
+  // This needs to occur at the end of this function
+  iterator.setDate(date + 1);
+
+  return (
+    <td className={className}>
+      {date}
+    </td>
+  );
+}
+
+function Calendar(props) {
+  const highlightedDate = new Date(props.value ? `${props.value}T00:00` : Date.now());
+  const [year, setYear] = useState(highlightedDate.getFullYear());
+  const [month, setMonth] = useState(highlightedDate.getMonth());
+  // const [date] = useState(highlightedDate.getDate());
+
+  function onPreviousMonthPress() {
+    const tmpDate = new Date(year, month - 1);
+    setMonth(tmpDate.getMonth());
+    setYear(tmpDate.getFullYear());
+  }
+
+  function onNextMonthPress() {
+    const tmpDate = new Date(year, month + 1);
+    setMonth(tmpDate.getMonth());
+    setYear(tmpDate.getFullYear());
+  }
+
+  const previousCalendarMonth = new Date(year, month - 1);
+  const currentCalendarMonth = new Date(year, month);
+  const nextCalendarMonth = new Date(year, month + 1);
+  const iterator = getDateIterator(year, month);
+  const headIterator = new Date(iterator.getTime());
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.header}>
+        <IconButton
+          className={styles['change-month-button']}
+          icon={arrowLeft}
+          label={`Change calendar to ${previousCalendarMonth.toLocaleDateString(undefined, {
+            year: 'numeric',
+            month: 'long',
+          })}`}
+          onClick={onPreviousMonthPress}
+        />
+        <button
+          className={styles['year-button']}
+        >
+          {currentCalendarMonth.toLocaleDateString(undefined, {
+            year: 'numeric',
+            month: 'long',
+          })}
+          <Icon
+            className={styles['change-year-icon']}
+            icon={caretDown}
+            v={2}
+          />
+        </button>
+        <IconButton
+          className={styles['change-month-button']}
+          icon={arrowRight}
+          label={`Change calendar to ${nextCalendarMonth.toLocaleDateString(undefined, {
+            year: 'numeric',
+            month: 'long',
+          })}`}
+          onClick={onNextMonthPress}
+        />
+      </div>
+      <table className={styles['calendar-grid']}>
+        <thead>
+          <tr>
+            {getHeadCell(headIterator)}
+            {getHeadCell(headIterator)}
+            {getHeadCell(headIterator)}
+            {getHeadCell(headIterator)}
+            {getHeadCell(headIterator)}
+            {getHeadCell(headIterator)}
+            {getHeadCell(headIterator)}
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            {getCell(iterator, month, highlightedDate)}
```

Comment:
It looks like we're copypasta-ing here. Could we extract this to a separate function that operates in the same way? Something like

```javascript
function foo() {
  const foo = [];

  for (let i = 0; i < 7; i++) {
    foo.push(getCell(iterator, month, highlightedDate));
  }

  return foo;
}

/* ... */

{ foo() }

Reply:
I'm okay with figuring out how to do this a little bit more concisely. However, I want to wait until we implement interactivity since we might need to add more logic around the rendering process.